runtime/j9vm/jvm.c

```diff
@@ -3991,10 +3998,35 @@ JVM_LoadLibrary(const char *libName, jboolean throwOnFailure)
 		JavaVM *vm = (JavaVM *)javaVM;
 		(*vm)->GetEnv(vm, (void **)&env, JNI_VERSION_1_2);
 		if (NULL != env) {
-			char errMsg[512];
-			jio_snprintf(errMsg, sizeof(errMsg), "Failed to load library \"%s\"", libName);
-			errMsg[sizeof(errMsg) - 1] = '\0';
+			char errBuf[512];
+			char *errMsg = errBuf;
+			int bufSize = sizeof(errBuf);
+			const char *portMsg = attemptedLoad ? j9error_last_error_message() : "";
+			if ('\0' != *portMsg) {
+				/* The port library message includes the file name which failed to load. */
```

Comment:
Have you verified this on all platforms (the file name is part of the message)?

Reply:
The code in [`ClassLoader`](https://github.com/eclipse-openj9/openj9/blob/master/jcl/src/java.base/share/classes/java/lang/ClassLoader.java#L2009) creates a message which includes the file name as well as the error message which suggests not all platforms are has helpful as Linux.

`JVM_LoadSystemLibrary()` doesn't include the error message in the exception it throws (but I don't see any uses of that function (in OpenJ9 builds)).

Reply:
AIX, Linux, Mac all include the name. z/OS does not, and I don't think Windows does either. I've pushed a change to include the name.